### PR TITLE
Extend ECS service create timeout

### PR DESCRIPTION
**Pull request by danielrbradley**
Use a transformation to set the create custom timeout.

Fixes https://github.com/pulumi/pulumi-awsx/issues/1104

### CODE REVIEW

**Comment by danielrbradley** [comment]
@t0yv0 does this also close #112 ?

**Comment by t0yv0** [comment]
It seems #112 is closed already, not sure.

File: examples/ts-lb-with-subnets/index.ts [hunk 1]
```diff
@@ -10,25 +11,41 @@ const lb = new awsx.lb.ApplicationLoadBalancer("nginx-lb", {
...
+                    },
+                ],
+            },
         },
     },
-});
+    {
+        transformations: [
+            (args) => {
+                if (args.type === "aws:ecs/service:Service") {
```
Comment (t0yv0) [inline examples/ts-lb-with-subnets/index.ts:38]: Ah, very interesting, so this is transforming the "invisible" resource that's not visible in the program. This is a great but not very obvious workaround.

**Review by t0yv0 — APPROVED** [review]

**Review by thomas11 — COMMENTED** [review]
Wouldn't it be cleaner to expose `timeouts` in p-aws and then use it as a normal property in awsx, [here](https://github.com/pulumi/pulumi-awsx/blob/tkappler/more-upgrade-tests/awsx/ecs/fargateService.ts#L60-L61)? This would fix the problem in other places as well, and allow p-aws users access to the timeout.

**Comment by t0yv0** [comment]
I like that could you throw that into an issue?

**Comment by danielrbradley** [comment]
@thomas11 this is already exposed in the AWS provider ... but as a resource _option_ rather than a resource _argument_. We replicate arguments, but options are meant to be propegated within components automatically by the framework, but customTimeouts hasn't been implemented.

**Comment by danielrbradley** [comment]
> It seems #112 is closed already, not sure.

Oops, missed a "1" ... I was meaning:
- https://github.com/pulumi/pulumi-awsx/issues/1112

**Comment by t0yv0** [comment]
I'd like to keep that open to represent possibly user-facing issue.

**Comment by t0yv0** [comment]
That's tempting.

**Pull request merged**